graphql_compiler/schema_merging/rename_schema.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+
+from graphql import build_ast_schema, parse
+from graphql.language.visitor import Visitor, visit
+
+from .utils import SchemaError, get_schema_data
+
+
+RenamedSchema = namedtuple(
+    'RenamedSchema', [
+        'schema_ast',  # type: Document, ast representing the renamed schema
+        'reverse_name_map',  # type: Dict[str, str], new type name to old type name
+        'reverse_root_field_map'  # type: Dict[str, str], new root field name to old root field
+    ]
+)
+
+
+def rename_schema(schema_string, rename_func=lambda name: name):
+    """Create a RenamedSchema, where types and query root fields are renamed using rename_func.
+
+    Args:
+        schema_string: string describing a valid schema that does not contain extensions
+        rename_func: callable that takes string to string, used to transform the names of
+                     types, interfaces, enums, and root fields. Defaults to identity
+
+    Returns:
+        RenamedSchema
+
+    Raises:
+        SchemaError if input schema_string does not represent a valid schema without extensions,
+            or if there are conflicts between the renamed types or root fields
+    """
+    # Check that the input string is a parseable and valid schema.
+    try:
+        ast = parse(schema_string)
+        build_ast_schema(ast)
```

Comment:
Does `build_ast_schema` do anything? I think it's only a getter function and does not mutate the ast.

Reply:
I put it here as a way to check the validity of the input string -- if build_ast_schema doesn't raise errors, that says a lot about the legality of the structure of the input. It definitely feels inelegant and I should put a comment explaining the purpose of the line if nothing else. I'm not sure if there's a better way to check if the input is a good ast though.